### PR TITLE
Port ghost battle from Fire Red.

### PR DESCRIPTION
**Pull request by Jaizu**
## Description
Marked as draft, this doesn't compile yet.
Started porting the ghost battle from Fire Red.
What I would like to get done:
- [ ] Port the ghost battle from Fire Red aka make it compile
- [ ] Follow feedback to fix edge cases that might be missing due to the expansion engine

What I don't want to deal with but would be nice if someone wants to do it after this is merged or if someone wants to take this PR:
- [ ] Add support for better conditions for the battle to happen rather than just map + item
- [ ] Support for more species than Marowak
- [ ] Make it work with different ghosts under different conditions

## **Discord contact info**
671

### CODE REVIEW

**Comment by Jaizu** [comment]
Currently broken, it works fine on a vanilla based game but the expansion handles some things differently, so help and reviews are appreciatied.
Some screenies
image

**Comment by agsmgmaster64** [comment]
From my own port of this battle type, there's some weird shenanigans going on with recoil still happening if you choose a recoil move for if you battle a Ghost without the Silph Scope. I managed to fix this by just adding a check for IS_BATTLE_TYPE_GHOST_WITHOUT_SCOPE to MOVEEND_RECOIL.

**Comment by Jaizu** [comment]
I don't plan to work on this soon tm, so if anyone wants to take it feel free to do it. If you want to cleanup PRs/Issues you can just close this and anyone can retake the branch in the future or whatever you want from it, I don't plan to delete the branch or anything. I recommend cloning it anyway in case something weird happens to my repo, I'm not very good at organizing things.
TLDR: Do whatever you want/need with the code and the PR, won't be working on it soon, sorry.
